tests/foreman/cli/test_computeresource_azurerm.py

```diff
@@ -528,3 +545,122 @@ def test_positive_azurerm_host_provisioned(
+
+        request.cls.compute_attrs = (
+            f'resource_group={self.rg_default},vm_size={self.vm_size}, '
+            f'username={module_azurerm_byos_finishimg.username}, '
+            f'ssh_key_data={settings.azurerm.ssh_pub_key}, platform={self.platform},'
+            f'script_command={"touch /var/tmp/test.txt"}, script_uris={AZURERM_FILE_URI},'
+            f'premium_os_disk={self.premium_os_disk}'
+        )
+        nw_id = module_azurerm_cr.available_networks()['results'][-1]['id']
+        request.cls.interfaces_attributes = (
+            f'compute_network={nw_id},compute_public_ip=Static, compute_private_ip=false'
+        )
+
+    @pytest.fixture(scope='class')
+    def class_host_ft(
+        self,
+        azurermclient,
+        module_azurerm_byos_finishimg,
+        module_azurerm_cr,
+        default_architecture,
+        module_domain,
+        module_location,
+        module_org,
+        default_os,
+        default_smart_proxy,
+        module_puppet_environment,
+    ):
+        """
+        Provisions the host on AzureRM with BYOS Image
+        Later in tests this host will be used to perform assertions
+        """
+        set_hammer_api_timeout()
+        skip_yum_update_during_provisioning(template='Kickstart default finish')
+        host = Host.create(
+            {
+                'name': self.hostname,
+                'compute-resource': module_azurerm_cr.name,
+                'compute-attributes': self.compute_attrs,
+                'interface': self.interfaces_attributes,
+                'location-id': module_location.id,
+                'organization-id': module_org.id,
+                'domain-id': module_domain.id,
+                'architecture-id': default_architecture.id,
+                'operatingsystem-id': default_os.id,
+                'root-password': gen_string('alpha'),
+                'image': module_azurerm_byos_finishimg.name,
+                'volume': "disk_size_gb=5",
+            },
+            timeout=1800,
+        )
+        yield host
+        skip_yum_update_during_provisioning(template='Kickstart default finish', reverse=True)
+        Host.delete({'name': self.fullhostname}, timeout=1800)
+        set_hammer_api_timeout(reverse=True)
+
+    @pytest.fixture(scope='class')
+    def azureclient_host(self, azurermclient, class_host_ft):
+        """Returns the AzureRM Client Host object to perform the assertions"""
+        return azurermclient.get_vm(name=class_host_ft['name'].split('.')[0])
+
+    @upgrade
+    @tier3
+    def test_positive_azurerm_host_provisioned(
+        self, class_host_ft, azureclient_host, module_azurerm_cr, module_azurerm_byos_finishimg
+    ):
+        """Host can be provisioned on AzureRM using BYOS Image
+
+        :id: 5ebfc3ed-0e61-4cb1-9d5e-831d81bb3bcc
+
+        :CaseLevel: Component
+
+        ::CaseImportance: Critical
+
+        :steps:
+            1. Create a AzureRM Compute Resource with BYOS Image and provision host.
```

Comment:
You missed adding step saying `provision a host on that compute resource.`

Reply:
it's mentioned as `and provision host`  and good enough to understand as test is related to CR .